app/Models/Extensions/AlbumGetters.php

```diff
@@ -68,24 +68,30 @@ public function get_all_photos()
 			->where('albums._rgt', '<=', $this->_rgt);
 	}
 
-	public function get_thumbs()
+	public function get_thumb(): ?Thumb
 	{
-		[$sort_col, $sort_order] = $this->get_sort();
+		if ($this->cover != null) {
+			$cover = $this->cover;
+		} else {
+			[$sort_col, $sort_order] = $this->get_sort();
+
+			$sql = $this->get_all_photos();
+
+			//? apply safety filter : Do not leak pictures which are not ours
+			$forbiddenID = resolve(PublicIds::class)->getNotAccessible();
 
-		$sql = $this->get_all_photos();
+			if ($forbiddenID != null && !$forbiddenID->isEmpty()) {
+				$sql = $sql->whereNotIn('album_id', $forbiddenID);
+			}
 
-		//? apply safety filter : Do not leak pictures which are not ours
-		$forbiddenID = resolve(PublicIds::class)->getNotAccessible();
-		if ($forbiddenID != null && !$forbiddenID->isEmpty()) {
-			$sql = $sql->whereNotIn('album_id', $forbiddenID);
+			$cover = $sql->orderBy('star', 'DESC')
+				->orderBy($sort_col, $sort_order)
+				->orderBy('photos.id', 'ASC')
+				->limit(1)
```

Comment:
So are we eliminating the two "behind" cover pictures with this PR, leaving just the frontmost one?

Reply:
somewhat yes. I don't have strong opinion on that one. Visually, it will just be the same picture displayed 3 times.